File: Samples/VB/DriveInfo/Form1.vb
Public Class Form1

    Private Sub btnGetInfo_Click(ByVal sender As System.Object, ByVal e As System.EventArgs)
        'Get the drive details
        If Me.listBox1.SelectedIndex >= 0 Then
            Dim di As OpenNETCF.IO.DriveInfo = CType(Me.listBox1.SelectedValue, OpenNETCF.IO.DriveInfo)
            Me.AvailableFreeSpace.Text = di.AvailableFreeSpace.ToString("N0")
            Me.RootDirectory.Text = di.RootDirectory.ToString()
            Me.TotalFreeSpace.Text = di.TotalFreeSpace.ToString("N0")
            Me.TotalSize.Text = di.TotalSize.ToString("N0")
        End If

    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.listBox1.DataSource = OpenNETCF.IO.DriveInfo.GetDrives()
    End Sub
End Class

File: Samples/CSharp/DriveInfo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using OpenNETCF.IO;
namespace DriveInfo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            this.listBox1.DataSource = OpenNETCF.IO.DriveInfo.GetDrives();
        }

        private void btnGetInfo_Click(object sender, EventArgs e)
        {
            //Get the drive details
            if (this.listBox1.SelectedIndex >= 0)
            {
                OpenNETCF.IO.DriveInfo di = (OpenNETCF.IO.DriveInfo)this.listBox1.SelectedValue;
                this.AvailableFreeSpace.Text = di.AvailableFreeSpace.ToString("N0");
                this.RootDirectory.Text = di.RootDirectory.ToString();
                this.TotalFreeSpace.Text = di.TotalFreeSpace.ToString("N0");
                this.TotalSize.Text = di.TotalSize.ToString("N0");
            }
        
        }
    }
}